server/src/main/java/org/apache/druid/client/indexing/HttpIndexingServiceClient.java

```diff
@@ -70,7 +70,7 @@ public HttpIndexingServiceClient(
   public void killUnusedSegments(String idPrefix, String dataSource, Interval interval)
```

Comment:
I'm not sure what this class is used for since there arent any docs about it, but will adding the parameter here instead of defaulting to false break anything? If not, any reason not to add?

Reply:
The two places where it's used currently:

```
  @DELETE
  @Path("/{dataSourceName}/intervals/{interval}")
```
on the coordinator which just issues a kill task

and internally by the coordinator in a coordinator duty which periodically issues a kill task for segments that are unused (if that functionality is enabled in the coordinator).

For the former case, it might make sense to add this parameter, but the same API set already has the "markUnused" API, so I didn't feel it was necessary and wanted to limit the scope of the PR, and for the latter case, the "markUnused" flag doesn't make sense.